runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -248,7 +258,11 @@ private void findAndSetLogHead() {
     }
 
     private void findAndSetLogTail() {
-        logTail = runtime.getSequencerView().query().getTokenValue();
+        Layout newLayout = runtime.getLayoutView().getLayout();
```

Comment:
this logic (i.e. get last segment) seems to be in multiple places, can we push it to layout ?

Reply:
Done